processor/attributekeyprocessor/testdata/config.yaml

```diff
@@ -0,0 +1,27 @@
+receivers:
+  examplereceiver:
+
+processors:
+  attributes:
+    key-replacements:
+      foo: 
```

Comment:
add a test with key containing '.', this is a special case for viper, likely you will need to manually call UnmarshalKey on the map field instead of delegating it